dask_searchcv/adaptive.py

```diff
@@ -0,0 +1,446 @@
+            raise ValueError('n_jobs must be either -1 or 0')
+        self.n_jobs = n_jobs
+
+        est = model if not isinstance(model, ParallelPostFit) else model.estimator
+        if not hasattr(est, 'partial_fit'):
+            raise ValueError('Hyperband relies on partial_fit. Without it '
+                             'it would be no different than RandomizedSearchCV')
+        if not hasattr(est, 'warm_start'):
+            warnings.warn('model has no attribute warm_start. Hyperband will assume it '
+                          'is reusing previous calls to `partial_fit` during each call '
+                          'to `partial_fit`')
+        else:
+            if not est.warm_start:
+                raise ValueError('Hyperband relies on model(..., warm_start=True).')
+
+        self.history = []
+
+        super_default = dict(scoring=None, iid=True, refit=True, cv=None,
+                             error_score='raise', return_train_score=_RETURN_TRAIN_SCORE_DEFAULT,
+                             scheduler=None, cache_cv=True)
+        for k, v in super_default.items():
+            if k in kwargs and kwargs[k] != super_default[k]:
+                warnings.warn('Hyperband ignores the keyword argument {k}.')
+        super(Hyperband, self).__init__(model, **kwargs)
+
+    def fit(self, X, y, dry_run=False, **fit_kwargs):
+        """
+        This function requires an active dask.distribtued client as it uses
+        the dask.distributed's concurrent.futures API.
+
+        This function implicitly assumes that higher scores are better. Note:
+        this matters if you are using a scoring function that measures loss,
+        where lower values imply a better model.
+
+        This function works with classes with a certain API, not only
+        scikit-learn estimators. It is assumed that the class
+
+        * implements ``partial_fit(self, X, y, **kwargs)``.
+        * implements ``set_params(self, **kwargs) -> self``
+        * implements ``score(self, X, y) -> number``
+        * supports ``sklearn.base.clone(self, safe=True)``. Support for this
+          function comes down to having an function ``get_params(self)``.
+
+        """
+        import distributed
+        client = _get_client()
+
+        variables = {'val_score': -np.inf, 'model': None, 'config': None,
+                     'eta': self.eta, 'history': []}
+        shared = {}
+        for key, value in variables.items():
+            shared[key] = distributed.Variable(key)
+            shared[key].set(value)
+
+        if '_dask_hyperband_data' in client.list_datasets():
+            client.unpublish_dataset('_dask_hyperband_data')
+
+        X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.1)
+        data = {'train': [X_train, y_train], 'val': [X_val, y_val]}
+
+        for key in data.keys():
+            data[key][0] = client.scatter(data[key][0])
+            data[key][1] = client.scatter(data[key][1])
```

Comment:
I believe that dask dives within dicts.  Although it may not dive within deeper structures though, so actually my suggestion above may not be appropriate.  You might need something like the following:

    L = client.scatter(sum([[v[0], v[1]] for v in data.values()], []))
    data = {...}